cmd/buildah/images.go

```diff
@@ -142,10 +142,26 @@ func imagesCmd(c *cobra.Command, args []string, iopts *imageResults) error {
 		options.Filters = append(options.Filters, "intermediate=false")
 	}
 
-	images, err := runtime.ListImages(ctx, args, options)
+	imagesByNames, err := runtime.ListImagesByNames(args)
 	if err != nil {
 		return err
 	}
+	imagesByFilter, err := runtime.ListImages(ctx, options)
+	if err != nil {
+		return err
+	}
+
+	imagesByNameIDs := map[string]bool{}
+	for _, image := range imagesByNames {
+		imagesByNameIDs[image.ID()] = true
+	}
+
+	images := make([]*libimage.Image, 0, len(imagesByNames))
+	for _, image := range imagesByFilter {
```

Comment:
I think this is incorrect in the case where `len(args) == 0`; the previous code would use all images returned by the filter.

[Intuitively, it seems to me that the `len(args) > 0` (`len(args) == 1`) case should be a conditional code path, always doing the `ListImagesByNames` does not help clarity. But that’s a guess without trying.]